commands/errors.go

```diff
@@ -0,0 +1,55 @@
+package commands
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+// FindEntitySummary is thrown when at least, one error is found
+// when finding entities.
+type FindEntitySummary struct {
+	Errors []error
+}
+
+// ErrEntityNotFound is thrown when an entity (user, team, etc.)
+// is not found, returning the id sent by arguments
+type ErrEntityNotFound struct {
+	Type string
+	ID   string
+}
+
+func (e ErrEntityNotFound) Error() string {
+	return fmt.Sprintf("%s %s not found", e.Type, e.ID)
+}
+
+type NotFoundError struct {
+	Msg string
+}
+
+func (e *NotFoundError) Error() string {
+	return e.Msg
+}
+
+type BadRequestError struct {
+	Msg string
+}
+
+func (e *BadRequestError) Error() string {
+	return e.Msg
+}
+
+// ExtractErrorFromResponse extracts the error from the response,
+// encapsulating it if matches the common cases, such as when it's
+// not found, and when we've made a bad request
+func ExtractErrorFromResponse(r *model.Response) error {
+	switch r.Error.StatusCode {
+	case http.StatusNotFound:
+		return &NotFoundError{Msg: r.Error.Error()}
+	case http.StatusBadRequest:
+		return &BadRequestError{Msg: r.Error.Error()}
+	default:
```

Comment:
Should we have here an error case for insufficient permissions too?

Reply:
If it's a permission error the server should answer with a forbidden, right?

I'm making this function so I can ignore those two errors (bad request and not found) . I would leave it as it is until some other function manages it. What do you think?

Reply:
I think we can iterate and add more status codes if needed eventually, so I'm happy to move forward :)